### PR TITLE
DAOS-11741 test: CR Pass 3 - Corrupt label in MS

### PR DESCRIPTION
**Pull request by shimizukko**
Test steps:
1. Create a pool.
2. Mangle the label in the MS (Mangle the copy of the PS metadata exists in the MS).
3. Check that the label in MS is corrupted with -fault added.
4. Check that the label in PS isn’t corrupted.
5. Stop the servers and enable the checker.
6. Set the policy to --all-interactive.
7. Start the checker and query the checker until the fault is detected.
8. Repair with option 1; Trust PS pool label.
9. Query the checker until the fault is repaired.
10. Call dmg check disable and restart the servers.
11. Wait for all ranks to join.
12. Verify that the corrupted label in MS is fixed.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: test_corrupt_label_ms
Test-repeat: 6
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/recovery/pool_cleanup.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        dmg_command.check_disable()
+
+        # 11. Wait for all ranks to join.
+        self.log_step("Wait for all ranks to join.")
+        rank_list = self.server_managers[0].get_host_ranks(hosts=self.hostlist_servers)
+        failed_ranks = self.server_managers[0].check_rank_state(
+            ranks=rank_list, valid_states="joined", max_checks=10)
+        if failed_ranks:
+            self.fail(
+                f"Following ranks didn't become joined after restart! {failed_ranks}")
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/pool_cleanup.py:147]: Should this use `errors.append` to match above? What if there were already errors before we get here?

Comment (shimizukko) [reply src/tests/ftest/recovery/pool_cleanup.py]: Done

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Comment by ashleypittman** [comment]
> Pylint must pass in able for this to land: https://github.com/daos-stack/daos/actions/runs/6179044096/job/16773298623?pr=13029

Whilst that's technically true I think pylint has been failing on this branch for a while so it's not a regression from this PR.  It should be fixed however.

**Comment by phender** [comment]
> Whilst that's technically true I think pylint has been failing on this branch for a while so it's not a regression from this PR. It should be fixed however.

Right, it was presumably missed in gatekeeping of some other PR, but we should fix it now as effort was made to clean up the pylint issues in the master branch and we shouldn't be introducing any new pylint issues.

**Comment by daltonbohning** [comment]
> > Whilst that's technically true I think pylint has been failing on this branch for a while so it's not a regression from this PR. It should be fixed however.
> 
> Right, it was presumably missed in gatekeeping of some other PR, but we should fix it now as effort was made to clean up the pylint issues in the master branch and we shouldn't be introducing any new pylint issues.

If this PR is otherwise ready to land, I think it's better to fix in a separate PR since this one didn't even touch the file that's failing.

**Comment by daltonbohning** [comment]
Also, FWIW, the pylint was introduced by a direct merge commit from master:
https://github.com/daos-stack/daos/commit/31baa793a40fbc7ae3c7650497ed28b9fdb0dff4#

**Review by phender — APPROVED** [review]

**Comment by shimizukko** [comment]
We will bump up the maximum number of lines in a file, so we don't need to fix this pylint here.

**Pull request merged**